bootstrap.py

```diff
@@ -1053,12 +1066,78 @@ def exec_service(service, command, failonerror=True):
         migrate_systems(options.org, options.activationkey)
         if options.enablerepos:
             enable_repos()
+    elif options.new_capsule and options.foreman_fqdn:
+        # > ELIF new_capsule and foreman_fqdn set, will migrate to other capsule
+        #
+        # > will replace CA certificate, reinstall katello-agent, gofer
+        # > will optionally update hostgroup and location
+        # > wil update system definition to point to new capsule for content,
+        # > Puppet, OpenSCAP and update Puppet configuration (if applicable)
+        # > MANUAL SIGNING OF CSR OR MANUALLY CREATING AUTO-SIGN RULE STILL REQUIRED!
+        # > API doesn't have a public endpoint for creating auto-sign entries yet!
+
+        # Make system ready for switch, gather required data
+        get_bootstrap_rpm(clean=True, unreg=False)
+        install_katello_agent()
+        API_PORT = get_api_port()
+        capsule_id = return_matching_foreman_key('smart_proxies', 'name="%s"' % options.foreman_fqdn, 'id', False)
+        host_id = return_matching_foreman_key('hosts', 'name="%s"' % FQDN, 'id', False)
+        capsule_features = [f['name'] for f in get_json("https://" + options.foreman_fqdn + ":" + API_PORT + "/katello/api/capsules/%s" % capsule_id)['features']]
+
+        # Optionally configure new hostgroup, location
+        if options.hostgroup:
+            print_running("Calling Foreman API to switch hostgroup for %s to %s" % (FQDN, options.hostgroup))
+            hostgroup_id = return_matching_foreman_key('hostgroups', 'title="%s"' % options.hostgroup, 'id', False)
+            jdata = json.loads('{"host": {"hostgroup_id": "%s"}}' % hostgroup_id)
+            put_json("https://" + options.foreman_fqdn + ":" + API_PORT + "/api/hosts/%s" % host_id, jdata)
+        if options.location:
+            print_running("Calling Foreman API to switch location for %s to %s" % (FQDN, options.location))
+            location_id = return_matching_foreman_key('locations', 'title="%s"' % options.location, 'id', False)
+            jdata = json.loads('{"host": {"location_id": "%s"}}' % location_id)
+            put_json("https://" + options.foreman_fqdn + ":" + API_PORT + "/api/hosts/%s" % host_id, jdata)
+
+        # Configure new proxy_id for Puppet (if not skipped), and OpenSCAP (if available and not skipped)
```

Comment:
can we figure out which feature were enabled on the old proxy and only try to change those? relying on the cli options seems wrong here.

Reply:
This is always going to be a chicken and egg problem: even if Puppet is enabled for a system on the Foreman side, it doesn't mean the Puppet agent is actually configured and running, for example. 

For OpenSCAP I check for whether or not that's enabled on the Satellite^HForeman side.  Setting the openscap_proxy_id for a system doesn't actually do anything if no policy is assigned. If a policy *is* assigned, you want it set, if there is no policy assigned, it doesn't matter. OpenSCAP is only skipped if foreman is in options.skip, or when the new Capsule doesn't have it.

People might be running Puppet outside of the scope of Foreman, so the skip option for Puppet needs to be there. 

The Katello functionality needs always be done.

So, adding all up, I would leave the options, unless I'm overlooking something.